core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -174,31 +177,33 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
     }
 
     if (requests.isEmpty) {
-      // no outstanding requests so we don't expect any messages back, but we should anyway
-      // drive the KafkaConsumer by polling
-
-      def checkNoResult(rawResult: ConsumerRecords[K, V]): Unit =
-        if (!rawResult.isEmpty)
-          throw new IllegalStateException(s"Got ${rawResult.count} unexpected messages")
-
-      checkNoResult(consumer.poll(0))
-
-      // For commits we try to avoid blocking poll because a commit normally succeeds after a few
-      // poll(0). Using poll(1) will always block for 1 ms, since there are no messages.
-      // Therefore we do 10 poll(0) with short 10 μs delay followed by 1 poll(1).
-      // If it's still not completed it will be tried again after the scheduled Poll.
-      var i = 10
-      while (i > 0 && commitsInProgress > 0) {
-        LockSupport.parkNanos(10 * 1000)
-        val pollTimeout = if (i == 1) 1L else 0L
-        checkNoResult(consumer.poll(pollTimeout))
-        i -= 1
+      try {
+        // no outstanding requests so we don't expect any messages back, but we should anyway
+        // drive the KafkaConsumer by polling
+
+        def checkNoResult(rawResult: ConsumerRecords[K, V]): Unit =
+          if (!rawResult.isEmpty)
+            throw new IllegalStateException(s"Got ${rawResult.count} unexpected messages")
+
+        checkNoResult(consumer.poll(0))
+
+        // For commits we try to avoid blocking poll because a commit normally succeeds after a few
+        // poll(0). Using poll(1) will always block for 1 ms, since there are no messages.
+        // Therefore we do 10 poll(0) with short 10 μs delay followed by 1 poll(1).
+        // If it's still not completed it will be tried again after the scheduled Poll.
+        var i = 10
+        while (i > 0 && commitsInProgress > 0) {
+          LockSupport.parkNanos(10 * 1000)
+          val pollTimeout = if (i == 1) 1L else 0L
+          checkNoResult(consumer.poll(pollTimeout))
+          i -= 1
+        }
       }
+      finally wakeupTask.cancel()
```

Comment:
it will notify all stages that have requested messages, that's what I added in postStop elsewhere
for Kafka offsets, it's no different than any other crash/restart